src/sys/signal.rs

```diff
@@ -370,6 +372,15 @@ impl SigSet {
         try!(pthread_sigmask(how, Some(self), Some(&mut oldmask)));
         Ok(oldmask)
     }
+
+    /// Suspends execution of the calling thread until one of the signals in the
+    /// signal mask becomes pending, and returns the accepted signal.
+    pub fn wait(&self) -> Result<SigNum> {
+        let mut signum: SigNum = unsafe { mem::uninitialized() };
```

Comment:
What is present is valid and marginally more performant than initializing to zero.  It is analogous to what would be done in C.  We just need a spot on the stack that the kernel will write to.